drivers/media/i2c/mira220.c

```diff
@@ -0,0 +1,2481 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * A V4L2 driver for ams MIRA220 cameras.
+ * Copyright (C) 2022, ams-OSRAM
+ *
+ * Based on Sony IMX219 camera driver
+ * Copyright (C) 2019, Raspberry Pi (Trading) Ltd
+ */
+
+#include <linux/clk.h>
+#include <linux/delay.h>
+#include <linux/gpio/consumer.h>
+#include <linux/i2c.h>
+#include <linux/module.h>
+#include <linux/pm_runtime.h>
+#include <linux/regulator/consumer.h>
+#include <media/v4l2-ctrls.h>
+#include <media/v4l2-device.h>
+#include <media/v4l2-event.h>
+#include <media/v4l2-fwnode.h>
+#include <media/v4l2-mediabus.h>
+#include <media/v4l2-cci.h>
+#include <linux/unaligned.h>
+
+/*
+  * Introduce new v4l2 control
+  */
+#include <linux/v4l2-controls.h>
+
+#define MIRA220_NATIVE_WIDTH 1600U
+#define MIRA220_NATIVE_HEIGHT 1400U
+
+#define MIRA220_PIXEL_ARRAY_LEFT 0U
+#define MIRA220_PIXEL_ARRAY_TOP 0U
+#define MIRA220_PIXEL_ARRAY_WIDTH 1600U
+#define MIRA220_PIXEL_ARRAY_HEIGHT 1400U
```

Comment:
Looking at https://look.ams-osram.com/m/2feacfd5e2a2d20e/original/Mira220-1-2-7-2-2MP-NIR-Enhanced-Global-Shutter-Image.pdf, Figure 1 says the sensor has 32 extra rows top and bottom, and 21 extra columns left and right.